x/zoneconcierge/keeper/ibc_packet_btc_timestamp.go

```diff
@@ -148,10 +148,47 @@ func (k Keeper) getDeepEnoughBTCHeaders(ctx context.Context) []*btclctypes.BTCHe
 	return k.btclcKeeper.GetMainChainFrom(ctx, startHeight)
 }
 
-// getHeadersToBroadcast retrieves headers to be broadcasted to all open IBC channels to ZoneConcierge
-// The header to be broadcasted are:
-// - either the whole known chain if we did not broadcast any headers yet
-// - headers from the child of the most recent header we sent which is still in the main chain up to the current tip
+// getHeadersToBroadcastForConsumer retrieves headers to be broadcasted to a specific Consumer
+// The headers to be broadcasted are:
+// - If no Consumer base header exists: use the last k+1 headers from tip (fallback)
+// - If Consumer base header exists but no headers sent yet: from Consumer base to tip
+// - If headers previously sent: from child of most recent valid header to tip
+// - If reorg detected: from Consumer base to tip
+func (k Keeper) getHeadersToBroadcastForConsumer(ctx context.Context, consumerID string) []*btclctypes.BTCHeaderInfo {
+	baseHeader := k.GetConsumerBaseBTCHeader(ctx, consumerID)
+	lastSegment := k.GetLastSentSegment(ctx)
+
+	// If no Consumer base header exists, fallback to the old behavior
+	if baseHeader == nil {
+		return k.getHeadersToBroadcast(ctx)
+	}
+
+	// If we haven't sent any headers yet, send from Consumer base to tip
+	if lastSegment == nil {
+		return k.btclcKeeper.GetMainChainFrom(ctx, baseHeader.Height+1)
+	}
```

Comment:
It might be possible that the BSN chooses a super old base BTC header such that L168 will get a large number of BTC headers. This might cause unexpected behaviours.

Since it's BSN's responsibility to pick a recent header, what do you think we add a restriction here, e.g., if `tip.Height - baseHeader.Height > k` then we return error?